analysis_templates/cms_minimal/__cf_module_name__/weight/example.py

```diff
@@ -0,0 +1,43 @@
+# coding: utf-8
+
+"""
+Example event weight producer.
+"""
+
+from columnflow.weight import WeightProducer, weight_producer
+from columnflow.util import maybe_import
+from columnflow.config_util import get_shifts_from_sources
+from columnflow.columnar_util import Route
+
+ak = maybe_import("awkward")
+np = maybe_import("numpy")
+
+
+@weight_producer(
```

Comment:
I suppose we could have done this with a Producer that stores a certain `weight` column, but similarly to Categorizer, I feel like the purpose of this particular type of Producer is much clearer by allowing it to return the weight directly.

Reply:
Fair point! I guess if we used a simple `Producer` there would be no straightforward way to specify which one of the produced columns should be used as a weight when filling histograms, and one would instead need to rely on a naming convention like using `weight`. In the `WeightProducer` approach, the computed weight is returned alongside the `events` so it can be used directly without needing to go through a column.